flang/lib/Lower/CallInterface.cpp

```diff
@@ -681,9 +708,27 @@ class Fortran::lower::CallInterfaceImpl {
   }
 
   void handleExplicitResult(
-      const Fortran::evaluate::characteristics::FunctionResult &) {
-    TODO(interface.converter.genLocation(),
-         "lowering interface with result requiring explicit interface");
+      const Fortran::evaluate::characteristics::FunctionResult &result) {
+    using Attr = Fortran::evaluate::characteristics::FunctionResult::Attr;
+
+    if (result.IsProcedurePointer())
+      TODO(interface.converter.genLocation(),
+           "procedure pointer result not yet handled");
+    const auto *typeAndShape = result.GetTypeAndShape();
+    assert(typeAndShape && "expect type for non proc pointer result");
+    auto mlirType = translateDynamicType(typeAndShape->type());
+    fir::SequenceType::Shape bounds = getBounds(typeAndShape->shape());
+    if (!bounds.empty())
+      mlirType = fir::SequenceType::get(bounds, mlirType);
+    if (result.attrs.test(Attr::Allocatable))
+      mlirType = fir::BoxType::get(fir::HeapType::get(mlirType));
+    if (result.attrs.test(Attr::Pointer))
+      mlirType = fir::BoxType::get(fir::PointerType::get(mlirType));
+    addFirOutput(mlirType, FirPlaceHolder::resultEntityPosition,
```

Comment:
Output sounds like I/O.

Reply:
Renamed to `addFirResult`/`addFirOperand`.

Reply:
Great. That's clearer.

Isn't everything "Fir" in some way though?